atc/exec/set_pipeline_step.go

```diff
@@ -105,7 +105,42 @@ func (step *SetPipelineStep) Run(ctx context.Context, state RunState) error {
 		return nil
 	}
 
-	team := step.teamFactory.GetByID(step.metadata.TeamID)
+	var team db.Team
+	if step.plan.Team == "" {
+		team = step.teamFactory.GetByID(step.metadata.TeamID)
+	} else {
+		currentTeam, found, err := step.teamFactory.FindTeam(step.metadata.TeamName)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return fmt.Errorf("team %s not found", step.metadata.TeamName)
+		}
+
+		targetTeam, found, err := step.teamFactory.FindTeam(step.plan.Team)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return fmt.Errorf("team %s not found", step.plan.Team)
+		}
+
+		permitted := false
+		if targetTeam.ID() == currentTeam.ID() {
+			permitted = true
+		}
+		if currentTeam.Admin() {
+			permitted = true
+		}
+		if !permitted {
```

Comment:
I think this could be simplified to

```
if !currentTeam.Admin() {
```

and not allow non-admin teams to specify a team

Reply:
Resolved in  ac6d54a